tests/xdsl_opt/test_xdsl_opt.py

```diff
@@ -7,7 +7,12 @@
 from xdsl.ir import MLContext
 from xdsl.passes import ModulePass
 from xdsl.utils.exceptions import DiagnosticException
-from xdsl.xdsl_opt_main import xDSLOptMain
+from xdsl.xdsl_opt_main import get_all_dialects, get_all_passes, xDSLOptMain
+
+
+def test_dialects_and_passes():
+    assert len(get_all_dialects()) > 0
```

Comment:
sorry, I meant something like this:
```suggestion
    names = [d.__name__ for d in get_all_dialects()]
    assert names == sorted(names), "Names of dialects should be in alphabedital order"
```

Reply:
Oh no, I don't think we want to have that!
I just want it in the function definition to make it cleaner!

Reply:
Why not?

Reply:
It would automate this little bit of the process, I don't see the harm in it TBH

Reply:
It feels like an arbitrary invariant to me, though I don't see how it could be an issue atm, given that dialects import their dependencies aswell...

But the more invariants, the more people rely on them and the harder to change they become.

I don't have a strong opinion, I just see a bigger disadvantage than the benefit we get.

Reply:
I also don't have a strong opinion, it's just that comments that say "please" in them aren't followed as closely as those that have a matching test to catch transgressors :)

Reply:
Yeah I would be against it to be honest.
The harm is that we are adding additional constraints on the API that we don't need, and that may change in the future. So I would prefer not introducing them in the first place.
Also, this check won't work, as dialects are object, not classes, so `__name__` won't exist anyway!

Reply:
I did not see the comment before, IMO, we should remove that :sweat_smile:

Reply:
@webmiche, do you mean we should remove the comment about keeping them in order?
I added it only because I felt it should be sorted, both for readability and conficts.
But if this cause too much trouble I'm fine removing it

Reply:
Hmm not sure. I would remove the comment, but I am fine with leaving it in, as long as we do not actually enforce it.